rc-events-service/src/test/scala/com/epam/random_coffee/events/services/EventServiceImplSpec.scala

```diff
@@ -37,20 +44,50 @@ class EventServiceImplSpec extends AsyncWordSpec with AsyncMockFactory with OneI
     }
 
     "update an event" when {
-      "event exists" in {
-        (repo.get _).expects(id).returns(Future.successful(Some(event)))
+      "event exists and request contains all fields" in {
+        (repo.get _).expects(id).returns(Future.successful(Some(rCEvent)))
+
+        (repo.update _).expects(*).returns(Future.unit)
+
+        service.update(id, Some(rCEvent.name), Some(rCEvent.description), Some(rCEvent.eventDate)).map(_ => succeed)
+      }
+
+      "request contains only the name" in {
+        (repo.get _).expects(id).returns(Future.successful(Some(rCEvent)))
+
+        (repo.update _).expects(*).returns(Future.unit)
+
+        service.update(id, Some(rCEvent.name), None, None).map(_ => succeed)
```

Comment:
This test checks nothing. You should at least check that service returns an event with the updated name (optionally you could check that it's calling a repository with an updated object).
Same about other tests.

Reply:
Sorry for these absolute useless test, I rewrote it.